### PR TITLE
Verifying (and fixing) issue 938

### PR DESCRIPTION
**Pull request by lemire**
Verifies https://github.com/simdjson/simdjson/issues/938

Fixes https://github.com/simdjson/simdjson/issues/938

### CODE REVIEW

**Comment by lemire** [comment]
@jkeiser Ok. So I think that there was a logical error in the code. We would reallocate the *implementation* and then never reallocate the memory for the document, as far as I can tell. I don't understand the logic there because I cannot see why you'd want to not reallocate the document buffer when you have reallocated the implementation memory.

I ended up just deleting code that I don't understand.

This might be a Chesterton's fence. Can you have a look? Maybe you had some reason for this code that I don't see right now.

**Comment by jkeiser** [comment]
@lemire the reason for the if statement is that people could change max_depth *without* changing capacity, and then we don't want to reallocate things we don't have to. It's not a huge problem in the grand scheme of things, but I was trying to be delicate.

I don't inherently mind this, but I think the lowest-touch fix here is to move the if () and document allocation up *above* the implementation allocation. The issue is that capacity gets updated during implementation allocation and so the if () statement will return true even though we haven't reallocated yet.

**Comment by lemire** [comment]
> I don't inherently mind this, but I think the lowest-touch fix here is to move the if () and document allocation up above the implementation allocation. The issue is that capacity gets updated during implementation allocation and so the if () statement will return true even though we haven't reallocated yet.

Done.

**Comment by lemire** [comment]
@jkeiser  Hmmm...

**Comment by lemire** [comment]
Should be better now.

**Comment by lemire** [comment]
@jkeiser I'll wait for your approval.

File: include/simdjson/inline/parser.h [hunk 1]
```diff
@@ -151,15 +151,22 @@ inline error_code parser::allocate(size_t capacity, size_t max_depth) noexcept {
...
     err = implementation->allocate(capacity, max_depth);
   } else {
+    need_doc_allocation = true;
     err = simdjson::active_implementation->create_dom_parser_implementation(capacity, max_depth, implementation);
   }
   if (err) { return err; }
-
-  if (implementation->capacity() != capacity || !doc.tape) {
-    return doc.allocate(capacity);
+  if (need_doc_allocation) {
```
Comment (jkeiser) [inline include/simdjson/inline/parser.h:167]: I see why you did it this way. Okey doke.

**Pull request merged**